experiments/download_ssl4eo.py

```diff
@@ -0,0 +1,738 @@
+
+
+def get_period(date: datetime, days: int = 5) -> Tuple[str, str]:
+    date1 = date - timedelta(days=days / 2)
+    date2 = date + timedelta(days=days / 2)
+    return date2str(date1), date2str(date2)
+
+
+"""get collection and remove clouds from ee"""
+
+
+def maskS2clouds(image: ee.Image) -> ee.Image:
+    qa = image.select("QA60")
+    # Bits 10 and 11 are clouds and cirrus, respectively.
+    cloudBitMask = 1 << 10
+    cirrusBitMask = 1 << 11
+    # Both flags should be set to zero, indicating clear conditions.
+    mask = qa.bitwiseAnd(cloudBitMask).eq(0)
+    mask = mask.bitwiseAnd(cirrusBitMask).eq(0)
+    return image.updateMask(mask)
+
+
+def get_collection_s2a(cloud_pct: float = 20) -> ee.ImageCollection:
+    collection = ee.ImageCollection("COPERNICUS/S2_SR")
+    collection = collection.filter(ee.Filter.lt("CLOUDY_PIXEL_PERCENTAGE", cloud_pct))
+    # collection = collection.map(maskS2clouds)
+    return collection
+
+
+def get_collection_s2c(cloud_pct: float = 20) -> ee.ImageCollection:
+    collection = ee.ImageCollection("COPERNICUS/S2")
+    collection = collection.filter(ee.Filter.lt("CLOUDY_PIXEL_PERCENTAGE", cloud_pct))
+    # collection = collection.map(maskS2clouds)
+    return collection
+
+
+def get_collection_s1() -> ee.ImageCollection:
+    collection = ee.ImageCollection("COPERNICUS/S1_GRD")
+    return collection
+
+
+def filter_collection(
+    collection: ee.ImageCollection, coords: List[float], period: Tuple[str, str]
+) -> ee.ImageCollection:
+    filtered = collection
+    if period is not None:
+        filtered = filtered.filterDate(*period)  # filter time
+    filtered = filtered.filterBounds(ee.Geometry.Point(coords))  # filter region
+
+    if filtered.size().getInfo() == 0:
+        raise ee.EEException(
+            f"ImageCollection.filter: No suitable images found in \
+                ({coords[1]:.4f}, {coords[0]:.4f}) between {period[0]} and {period[1]}."
+        )
+    return filtered
+
+
+def filter_collection_s1(
+    collection: ee.ImageCollection, coords: List[float], period: Tuple[str, str]
+) -> ee.ImageCollection:
+    filtered = collection
+    if period is not None:
+        filtered = filtered.filterDate(*period)  # filter time
+    filtered = filtered.filterBounds(ee.Geometry.Point(coords))  # filter region
+
+    filtered = filtered.filter(
+        ee.Filter.listContains("transmitterReceiverPolarisation", "VV")
+    ).filter(ee.Filter.listContains("transmitterReceiverPolarisation", "VH"))
+    filtered = filtered.filter(ee.Filter.eq("instrumentMode", "IW"))
+    # filtered = filtered.filter(ee.Filter.eq('orbitProperties_pass', 'ASCENDING'))
+    # filtered = filtered.filter(ee.Filter.eq('orbitProperties_pass', 'DESCENDING'))
+
+    if filtered.size().getInfo() == 0:
+        raise ee.EEException(
+            f"ImageCollection.filter: No suitable images found in \
+            ({coords[1]:.4f}, {coords[0]:.4f}) between {period[0]} and {period[1]}."
```

Comment:
```suggestion
            "ImageCollection.filter: No suitable images found in "
            f"({coords[1]:.4f}, {coords[0]:.4f}) between {period[0]} and {period[1]}."
```
Otherwise you get a ton of extra spaces in the middle of your error msg